Project/src/main/java/app/FileUploadController.java

```diff
@@ -30,29 +26,77 @@
      */
     @GetMapping("/upload")
     public String getFileUploader(Model model) {
-        User user = super.getUser();
-        model.addAttribute("files", fileRepository.findByUser(user));
+        model.addAttribute("articles", articleRepository.findAll());
+        model.addAttribute("role", super.getUser().getRole().getClass());
         return "upload";
     }
 
+    @ModelAttribute("reviewers")
+    public List<User> populateReviewers() {
+        return userRepository.findByRole_Name(Role.reviewer);
+    }
+
     /***
      * Uploads a file from form data
      * @param file The Multipart file that got uploaded
      * @return
      * @throws IOException
      */
     @PostMapping("/upload")
-    public String uploadFile(@RequestParam("file") MultipartFile file) throws IOException {
-        File newFile = new File();
-        newFile.setFileName(file.getOriginalFilename());
-        newFile.setFileType(file.getContentType());
-        newFile.setData(file.getBytes());
-        newFile.setStatus(ArticleStatus.SUBMITTED);
-        List users = new ArrayList<User>();
-        users.add(super.getUser());
-        newFile.setUser(users);
-        fileRepository.save(newFile);
+    public String uploadFile(Model model, @RequestParam("file") MultipartFile file) throws IOException {
+        Article newArticle = new Article();
+        newArticle.setFileName(file.getName());
+        newArticle.setFileType(file.getContentType());
+        newArticle.setData(file.getBytes());
+        newArticle.addAuthorizedUser(super.getUser());
+        Article article = new Article();
+
+        articleRepository.save(newArticle);
+
+        return "redirect:/upload";
+    }
+
+    @GetMapping("/addReviewer")
+    public String addReviewer(@RequestParam("articleId") String id, @RequestParam("reviewer") String userId) {
+        User callingUser = super.getUser();
+        User reviewingUser = userRepository.findById(Long.parseLong(userId));
+        if (reviewingUser.getRole() instanceof Reviewer && callingUser.getRole() instanceof Editor) {
```

Comment:
This may need to be completely changed based on reviews on your design choice of roles

Reply:
Also document methods